src/pages/Swap/index.tsx

```diff
@@ -18,7 +18,12 @@ export function SwapPage() {
     return <SwapPageRedirect />
   }
 
-  return <SwapWidget />
+  return (
+    <>
+      <SwapDerivedStateUpdater />
```

Comment:
This can't be the fix, the fix is the `useSafeMemoObject` in derived state hook, right? 
cause the hook is rendered no matter if its in `SwapWidget` or in the updater

in any case, i like you moved it to the updater

Reply:
Actually, the key problem is a place where `useFillSwapDerivedState` is called. `useSafeMemoObject` is just additional improvement